Add xnnpack to llama runner mac & linux CI job

lucylq opened the pull request:
Summary: Remake of D55290786, which is ghstack poisoned

Differential Revision: D55349949

mcr229 commented on .github/workflows/pull.yml:
```diff
@@ -90,6 +90,7 @@ jobs:
       matrix:
         dtype: [fp32]
         build-tool: [buck2, cmake]
+        mode: [portable, xnnpack]
```
can we select quant mode rather than portable/xnnpack? Right now this is choosing xnnpack but its actually quantizing the model. Instead I think we should have quant mode be what's selected, and XNNPACK will be the quantized backend which runs it.

lucylq replied:
Let's update in follow on, as we also want the pt2e quant path.

mcr229 commented on examples/models/llama2/CMakeLists.txt:
```diff
@@ -54,21 +54,15 @@ find_package(executorch CONFIG REQUIRED)
-                      $<TARGET_FILE:optimized_native_cpu_ops_lib> \
-                      LINKER:--no-whole-archive")
+  list(APPEND link_libraries optimized_native_cpu_ops_lib optimized_kernels
+    portable_kernels)
```
portable kernels is linked regardless of optimized? Can we move this outside of the if?

lucylq replied:
Hmn, `optimized_native_cpu_ops_lib` contains portable and optimized kernel registrations. If we don't have an optimized version of an op that's used by the model, we take the portable one.

mcr229 commented on examples/models/llama2/CMakeLists.txt:
```diff
@@ -54,21 +54,15 @@ find_package(executorch CONFIG REQUIRED)
+  target_link_options_shared_lib(portable_ops_lib)
 endif()
 
 target_link_libraries(llama_main PUBLIC gflags llama_runner)
```
can we add executorch here as well? This lib dep contains prim ops which need to be registered via target_link_options_shared_lib

lucylq replied:
thanks - good catch

lucylq committed:
Add xnnpack to llama runner mac & linux CI job (#2677)

Summary:

Remake of D55290786, which is ghstack poisoned

Reviewed By: kimishpatel, digantdesai

Differential Revision: D55349949
